### PR TITLE
Suggest major versions changes

### PR DESCRIPTION
**Pull request by EyalDelarea**
- [x] All [tests](https://github.com/jfrog/frogbot#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
---

- Return the ability to suggest major version changes (upgrade/downgrade) to fix an impacted package,
- Implemented our optimal fix version suggestions algorithm.
- Log unsupported indirect dependencies fixes.
- Update Frogbot badge to JFrog Frogbot. ( notice it will take effect only after release )

depends on https://github.com/jfrog/gofrog/pull/39

### CODE REVIEW

File: commands/createfixpullrequests.go [hunk 1]
```diff
@@ -273,23 +273,27 @@ func (cfp *CreateFixPullRequestsCmd) addVulnerabilityToFixVersionsMap(vulnerabil
...
-		majorChanged := utils.IsMajorVersionChange(fixVersionCandidate, currVersionStr)
-		if majorChanged {
-			return ""
-		}
 		if currVersion.Compare(fixVersionCandidate) > 0 {
 			return fixVersionCandidate
 		}
 	}
+	// When no upgrade was found,search for downgrade options
+	for i := len(fixVersions) - 1; i >= 0; i-- {
```
Comment (eyalbe4) [inline commands/createfixpullrequests.go:290]: I suggest we change the "fix version selection" algorithm to be the "closest version".

**Review by eyalbe4 — CHANGES_REQUESTED** [review]

File: commands/utils/utils.go [hunk 2]
```diff
@@ -178,11 +185,80 @@ func IsDirectDependency(impactPath [][]formats.ComponentRow) (bool, error) {
...
 }
 
-// Assuming version comes in a.b.c format,without any prefixes.
-func IsMajorVersionChange(fixVersionCandidate string, currVersionStr string) bool {
-	index := 0
-	separator := "."
-	candidateMajorVersion := strings.Split(fixVersionCandidate, separator)[index]
-	currentMajorVersion := strings.Split(currVersionStr, separator)[index]
-	return candidateMajorVersion != currentMajorVersion
+// Accepts current impacted version and two of it's closest neighbours, upgrade and downgrade suggestions.
```
Comment (omerzi) [inline commands/utils/utils.go:188]: it's -> its

File: commands/utils/utils.go [hunk 3]
```diff
@@ -178,11 +185,80 @@ func IsDirectDependency(impactPath [][]formats.ComponentRow) (bool, error) {
...
+	return ""
+}
+
+// isVersionUpgrade returns the version string with the higher version number
+// at a specified Major, Minor or Patch indexes.
+func isVersionUpgrade(v1 string, v2 string, index int) string {
+	v1Parts := strings.Split(v1, ".")
+	v2Parts := strings.Split(v2, ".")
+	for i := 0; i < 3; i++ {
+		v1Part, _ := strconv.Atoi(v1Parts[i])
```
Comment (omerzi) [inline commands/utils/utils.go:230]:
```suggestion
		v1Part, err := strconv.Atoi(v1Parts[i])
		if err != nil {
		     return "", err
		}
```

File: commands/utils/utils.go [hunk 4]
```diff
@@ -178,11 +185,80 @@ func IsDirectDependency(impactPath [][]formats.ComponentRow) (bool, error) {
...
+		v2Part, _ := strconv.Atoi(v2Parts[i])
```
Comment (omerzi) [inline commands/utils/utils.go:231]:
```suggestion
		v2Part, err := strconv.Atoi(v2Parts[i])
		if err != nil {
		   return "", err
		}
```

Comment (omerzi) [reply commands/utils/utils.go:231]: return err on all usages of the `strconv.Atoi()`

**Review by omerzi — CHANGES_REQUESTED** [review]
- Most of the logic under 'isVersionUpgrade', and `isVersionDowngrade` is similar. You can extract it into a general function that will receive a function based on the logic that differs between `isVersionUpgrade` and `isVersionDowngrade`.

- You can use the Version struct we defined in the gofrog package and modify it with these added functionalities. Keep it generic since gofrog is a generic Go package not related to Frogbot. 

- Errors should be returned. We don't want to ignore them most of the time.

**Pull request closed without merging**